### ISSUE TITLE
Lost emits

### ISSUE THREAD

**Issue by Jacobh2** [miguelgrinberg/Flask-SocketIO#804]
Hi!

tl;dr: What is the correct way of emitting from a background thread when using eventlet?

I'm using Flask-SocketIO together with [socketIO-client-2](https://pypi.org/project/socketIO-client-2/) in a client-server setup to sync data from the client up to the server.

When the server receives a message, this needs to be ack:ed on the client so it knows that it has correctly been processed. The ack:ing is asynchronous, so I have a callback function which push a message-id to a queue (Let's call this `qA`). This queue is then processed by a dedicated thread and using
```
socketio.emit('ack', delivery_tag, room=client_sid)
```
to send back the ack from another thread. This dedicated thread is started using
```
Thread(target=background_tread).start()
````
The server is using eventlet and I'm creating the SocketIO using
```
socketio = SocketIO(app, async_mode='eventlet, ping_interval=30, manage_session=False)
```
The very first thing I do on the server side is:
```
import eventlet 
eventlet.monkey_patch() 
```
to make sure that I can use the python's normal Threds!

My problem is that in the logs on the server, I see that I receive messages and start to process them.
```
@socketio.on('receive')
def receive_value_from_client(json_data):
    # This is a non blocking method
    handle_message(json_data)
```
At a slightly later point in time they are commited, so the callback is called with the correct message id. This message id is then put into the `qA` and the dedicated process thread correctly pulls out the message id and then does the `socketio.emit(...` call. But this message is never received on my client.

I feel that I'm missing something about how to use threads together with this lib. Is the preferred way of starting new threads to use `socketio.start_background_task(target=background_thread)` instead of `Thread(target=background_tread).start()` even if I've called `eventlet.monkey_patch() `?
